docs/en/advanced_guides/data_flow.md

```diff
@@ -1 +1,85 @@
+
+### Data Preprocessor to Model
+
+Though drawn separately in the diagram [above](#overview-of-dataflow), data_preprocessor is a part of the model and thus can be found in [Model tutorial](https://mmsegmentation.readthedocs.io/en/dev-1.x/advanced_guides/models.html) at Seg DataPreprocessor chapter.
+
+The return value of Data Preprocessor is a dict, contains `inputs` and `data_samples`, `inputs` would be transferred to GPU and some additional metainfo like `pad_shape` and `padding_size` would be added to the `data_samples`. When transfer to the network, the dict would be unpacked to two values. The following pseudo-codes show the return value of data preprocessor and the input values of model.
+
+```python
+dict(
+    inputs=torch.Tensor,
+    data_samples=Optional[List[SegDataSample], None]
+)
+```
+
+```python
+class Network(BaseSegmentor):
+
+    def forward(self, inputs: torch.Tensor, data_samples: List[SegDataSample], mode: str):
+        pass
+```
+
+### Model output
+
+#### To Evaluator
+
+At the evaluation procedure, the inference results would be transferred to `Evaluator`. You might read the [evaluation document](https://mmsegmentation.readthedocs.io/en/dev-1.x/advanced_guides/evaluation.html) for more information about `Evaluator`.
+
+After inference, the [BaseSegmentor](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/mmseg/models/segmentors/base.py#L15) in MMSegmentation would do a simple post process to pack inference results, the segmentation logits produced by the neural network, segmentation mask after the `argmax` operation and ground truth(if exists) would be packed into a similar `SegDataSample` instance. The return value of [postprocess_result](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/mmseg/models/segmentors/base.py#L132) is a **`List` of `SegDataSample`**. Following diagram shows the key properties of these `SegDataSample` instances.
+
+![SegDataSample](../../../resources/SegDataSample.png)
+
+#### Optim Wrapper
+
+The same as Data Preprocessor, loss function is also a part of the model, it's a property of [decode head](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/mmseg/models/decode_heads/decode_head.py#L142).
+
+In MMSegmentation, the method [loss_by_feat](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/mmseg/models/decode_heads/decode_head.py#L291) of `decode_head` is an unified interface used to compute loss.
+
+Parameters:
+
+- seg_logits (Tensor): The output from decode head forward function.
+- batch_data_samples (List\[:obj:`SegDataSample`\]): The seg data samples. It usually includes information such as `metainfo` and `gt_sem_seg`.
+
+Returns:
+
+- dict\[str, Tensor\]: a dictionary of loss components
```

Comment:
The `train_step` transfers the loss into OptimWrapper to update the weights in model, please refer [`train_step`](https://github.com/open-mmlab/mmengine/blob/6af88783fb0c7556cfcdd09c7ad839f9572c3095/mmengine/model/base_model/base_model.py#L84) for more details.